### PR TITLE
[1.14] Start generating recipes in data run

### PR DESCRIPTION
**Pull request by temp1011**
This is just some initial work on some of the more annoying recipes to handle in json. Eventually it would probably be nice to have everything generated.

I'm not sure of the best way to move the recipes from generated to a directory where they'll be loaded by the game.

With these changes the total size of the generated directory (including tags etc) is about *?* lines of json. (will update this once all recipes are being generated)

### CODE REVIEW

File: src/main/java/forestry/storage/features/BackpackItems.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 public class BackpackItems {
 	private static final IFeatureRegistry REGISTRY = ModFeatureRegistry.get(Constants.MOD_ID).getRegistry(ForestryModuleUids.BACKPACKS);
 
+	//TODO there is a race condition here: If Backpacks loads before apiculture then the group is null.
```
Comment (temp1011) [inline src/main/java/forestry/storage/features/BackpackItems.java]: I'm not 100% on the cause but there's definitely something breaking due to `ItemGroups` with these recipes.

Comment (Nedelosk) [reply src/main/java/forestry/storage/features/BackpackItems.java]: This should be fixed now.
https://github.com/ForestryMC/ForestryMC/blob/23b9a3ca8c24ecf6548a5898f0c102c97c802de8/src/main/java/forestry/core/ItemGroupForestry.java#L39-L40

Comment (temp1011) [reply src/main/java/forestry/storage/features/BackpackItems.java]: The problem still exists. If I put breakpoints on the static initialiser in `ItemGroupForestry` and `BackpackItems` then `BackpackItems` is loaded first. But I'm not really sure how to solve this without making the ItemGroups static initialisers, which means we can't hide items when the module isn't enabled.

Comment (Nedelosk) [reply src/main/java/forestry/storage/features/BackpackItems.java]: The easiest fix would be to give @FeatureProvider a priority value. And priorities ItemGroupForestry.

Comment (Nedelosk) [reply src/main/java/forestry/storage/features/BackpackItems.java]: https://github.com/ForestryMC/ForestryMC/blob/f02df303c2bbd66ab13ef8c212747a845c423755/src/main/java/forestry/modules/CommonModuleHandler.java#L99-L100

Comment (temp1011) [reply src/main/java/forestry/storage/features/BackpackItems.java]: It works perfectly now, thanks.

File: src/main/java/forestry/Forestry.java [hunk 2]
```diff
@@ -199,12 +201,16 @@ public void setupClient(FMLClientSetupEvent event) {
 	}
 
 	private void gatherData(GatherDataEvent event) {
+		CapabilityFluidHandler.register();	//TODO test
```
Comment (temp1011) [inline src/main/java/forestry/Forestry.java:226]: Forge calls this event in preInit which is either too late or not called on the data run. So to be able to create recipes with containers I added a call here. Probably forge should fix this.

File: src/main/java/forestry/core/data/ForestryRecipeProvider.java [hunk 3]
```diff
@@ -0,0 +1,1050 @@
...
+			return constructor.newInstance(stack);
+
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+
+	}
+
+
+	private void registerFluidsRecipes(RecipeDataHelper helper) {
```
Comment (temp1011) [inline src/main/java/forestry/core/data/ForestryRecipeProvider.java:1058]: The constructor of `IngredientNBT` isn't public so I have to use reflection to get to it. Again this seems like a forge issue.

**Review by temp1011 — COMMENTED** [review]
Note of hacks so far

**Comment by temp1011** [comment]
@Nedelosk is it acceptable to make the recipes that depend on the gamemode constant output? Or do we want to keep that feature?

**Comment by Nedelosk** [comment]
I have no idea if anyone uses the game mode feature anymore.

**Comment by temp1011** [comment]
I have no idea either.

**Comment by temp1011** [comment]
I guess this issue mentions them: https://github.com/ForestryMC/ForestryMC/issues/2245

**Comment by temp1011** [comment]
Screenshot from 2019-09-26 23-52-11
Putting this here in case it turns out to be an issue

**Comment by Nedelosk** [comment]
I pulled this into the 1.15.2 dev branch.

**Comment by thedarkcolour** [comment]
Did someone just interact with the CLA assisstant?

**Pull request closed without merging**